src/AirshipBindings.NETStandard/AirshipBindings.NETStandard.iOS/Airship.cs

```diff
@@ -247,12 +249,27 @@ public void AssociateIdentifier(string key, string identifier)
 
         public void DisplayMessageCenter()
         {
-            UAMessageCenter.Shared().Display();
+            if (OnMessageCenterDisplay != null)
+            {
+                OnMessageCenterDisplay.Invoke(this, new MessageCenterEventArgs());
+            }
+            else
+            {
+                UAMessageCenter.Shared().Display();
+            }
+  
         }
 
         public void DisplayMessage(string messageId)
         {
-            UAMessageCenter.Shared().DisplayMessage(messageId);
+            if (OnMessageCenterDisplay != null)
+            {
+                OnMessageCenterDisplay.Invoke(this, new MessageCenterEventArgs(messageId));
+            }
+            else
+            {
+                UAMessageCenter.Shared().DisplayMessage(messageId);
```

Comment:
Same idea on iOS except its a delegate that you will have to manage. I think this follows same pattern as the deep link listener.